### PR TITLE
implements ordering aggs and columns from admin dashboard

### PR DESCRIPTION
**Pull request by hlminh2000**
Some technical note:
- The initial idea of providing a preview using `react-grid-layout` introduces some additional complexities, we went with just rendering the name of the aggs.
- The main issue faced: `react-grid-layout` needs to be passed the dimentions of each element. This gets tricky as components are dynamic

@alex-wilmer @Jephuff

### CODE REVIEW

File: modules/components/src/Admin/Tabs/Aggregations/TableTab.js [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+        <div style={{ flex: 1 }}>
+          <EditColumns handleChange={columnsState.update} {...columnsState} />
+        </div>
+        <div>
+          <strong>Columns order</strong>
+          <DraggableListOrderer
+            {...{
+              itemsList: columnsState.state.columns.map(
+                ({ field, ...rest }) => ({
+                  field: field.split('.').join('__'),
```
Comment (Jephuff) [inline modules/components/src/Admin/Tabs/Aggregations/TableTab.js]: Why join with `__` here?

File: modules/components/src/Admin/Tabs/DraggableListOrderer.js [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+
+class DraggableOrders extends React.Component {
+  state = {
+    rowHeight: 1, // normalizes to pixel level
+    layout: [],
+    extendedMapping: [],
+  };
+  listElements = {};
+
+  adjustHeight = async listElements =>
```
Comment (Jephuff) [inline modules/components/src/Admin/Tabs/DraggableListOrderer.js]: Can you remove all of the resize logic and refs? just give everything a set height

File: modules/components/src/Admin/Tabs/DraggableListOrderer.js [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+      className = '',
+      style,
+    } = this.props;
+    const { extendedMapping } = this.state;
+    return (
+      <ReactGridLayout
+        className="layout"
+        cols={1}
+        layout={this.state.layout}
+        rowHeight={this.state.rowHeight}
```
Comment (Jephuff) [inline modules/components/src/Admin/Tabs/DraggableListOrderer.js]: This state never changes, could you just move the value inline?

File: modules/components/src/Admin/Tabs/DraggableListOrderer.js [hunk 4]
```diff
@@ -0,0 +1,154 @@
...
+              )?.displayName || item.field}
+            </div>
+          </div>
+        ))}
+      </ReactGridLayout>
+    );
+  }
+}
+
+export default ({
```
Comment (Jephuff) [inline modules/components/src/Admin/Tabs/DraggableListOrderer.js]: I think you can remove this wrapper, DraggableOrders can handle 0 items

File: modules/components/src/Admin/Tabs/Aggregations/AggregationsTab.js [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+    graphqlField={graphqlField}
+    render={aggsState => {
+      return (
+        <div
+          style={{
+            display: 'flex',
+            flexDirection: 'row',
+          }}
+        >
+          <div style={{ flex: 1 }}>
```
[the remark this answers is not in the document]
Comment (hlminh2000) [reply modules/components/src/Admin/Tabs/Aggregations/AggregationsTab.js]: EditAggs doesn't have a dom element wrapping it so the layout gets messed up :P

File: modules/components/src/Admin/Tabs/Aggregations/AggregationsTab.js [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+            <EditAggs handleChange={aggsState.update} {...aggsState} />
+          </div>
+          <div>
+            <strong>Aggs order</strong>
```
[the remark this answers is not in the document]
Comment (hlminh2000) [reply modules/components/src/Admin/Tabs/Aggregations/AggregationsTab.js]: yeah I'll go with `Aggregation Order` just to be extra specific cause the Aggregations and Table tabs look pretty similar

File: modules/components/src/Admin/Tabs/Aggregations/AggregationsTab.js [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+            <DraggableListOrderer
+              {...{
+                itemsList: aggsState.aggs.map(({ field, active, ...rest }) => ({
```
[the remark this answers is not in the document]
Comment (hlminh2000) [reply modules/components/src/Admin/Tabs/Aggregations/AggregationsTab.js]: artifacts of copy & paste :P thanks!

File: modules/components/src/Admin/Tabs/Aggregations/AggregationsTab.js (hunk 5 above)
[the remark this answers is not in the document]
Comment (Jephuff) [reply modules/components/src/Admin/Tabs/Aggregations/AggregationsTab.js]: those damn fragments haha
all good!

**Pull request merged**